packages/ui/src/components/surfaces/accordion/index.tsx

```diff
@@ -41,8 +50,30 @@ export const Accordion = ({
         [isControlled, expanded, internalExpanded]
     );
 
-    const summaryChildren = children[0];
-    const detailsChildren = children[1];
+    const { summaryChildren, detailsChildren } = React.Children.toArray(
```

Comment:
Spreading previous shouldn't be necessary in a reduce, we can just push. What about something like this?

```
const { summaryChildren, detailsChildren } = useMemo(() => {
        return React.Children.toArray(children).reduce(
            (
                accumulator: {
                    summaryChildren: ReactNode[];
                    detailsChildren: ReactNode[];
                },
                child
            ) => {
                if (matchChildByType(child, AccordionSummary))
                    accumulator.summaryChildren.push(child);
                else if (matchChildByType(child, AccordionDetails))
                    accumulator.detailsChildren.push(child);
                return accumulator;
            },
            { summaryChildren: [], detailsChildren: [] }
        );
    }, [children]);
```